components/ai_chat/core/common/utils.cc

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/common/utils.h"
+
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+
+#if !BUILDFLAG(IS_IOS)
+#include "content/public/browser/browser_context.h"
+#endif
+
+namespace ai_chat {
+
+namespace {
+
+bool IsDisabledByPolicy(PrefService* prefs) {
+  DCHECK(prefs);
+  return prefs->IsManagedPreference(prefs::kEnabledByPolicy) &&
+         !prefs->GetBoolean(prefs::kEnabledByPolicy);
+}
+
+#if !BUILDFLAG(IS_IOS)
+bool IsAllowedForContext(content::BrowserContext* context) {
+  return !context->IsTor();
+}
+#endif
+
+}  // namespace
+
+#if !BUILDFLAG(IS_IOS)
+bool IsAIChatEnabled(content::BrowserContext* context) {
+  DCHECK(context);
+  return IsAllowedForContext(context) &&
+         IsAIChatEnabled(user_prefs::UserPrefs::Get(context));
+}
+#endif
+
+bool IsAIChatEnabled(PrefService* prefs) {
```

Comment:
1. all these methods cannot be in `common` target. these are all `browser`-related stuff.
2. `common` can never reference `PrefService` and `content::`.

`common` is for things that are shared between `browser`, `renderer`, `ios`. Such as: features, command-line flags, mojom interfaces, generic helpers, etc.

Reply:
fixed in 314e3b1b065f49b21209355c424a5fd557bba091